### PR TITLE
linuxPackages.rtl88x2bu: 2020-08-20 -> 2021-01-21

### LINKED ISSUE #116315: rtl88x2bu fails to compile with kernel 5.10, 5.11
## Issue description

With this in `/etc/nixos/configuration.nix`:

```
  boot.kernelPackages = pkgs.linuxPackages_5_11;

  boot.extraModulePackages = [
    # asus usb wifi
    config.boot.kernelPackages.rtl88x2bu
  ]
```

I get compilation errors similar to https://github.com/NixOS/nixpkgs/issues/107829

## Fix

I was able to get things compiling (and WiFi working) by bumping the version and applying https://github.com/cilynx/rtl88x2bu/pull/132

```
  boot.kernelPackages = pkgs.linuxPackages_5_11;

  boot.extraModulePackages = [
    # asus usb wifi
    (config.boot.kernelPackages.rtl88x2bu.overrideAttrs (attrs: with pkgs; rec {
      name = "rtl88x2bu-${version}";
      version = "unstable-2021-01-20";
      src = fetchFromGitHub {
        owner = "cilynx";
        repo = "rtl88x2BU";
        # branch 5.6.1.6_35492.20191025_COEX20180928-6a6a @ 2021-01-20
        rev = "28b98b018a4a83059f10439b79fb5a2f98b618f6";
        sha256 = "077shn3na6spkr7qanrwmm2qyqsddxsxzbxnck3x2ijssnbbk6j3";
      };
      patches = [
        (fetchpatch {
          url = "https://github.com/cilynx/rtl88x2bu/pull/132.patch";
          sha256 = "0gbznxnwm517jsh3c1njb4qbxd9jskqviqzg545ww217q5063h2w";
        })
      ];
    }))
  ];
```

@Ralith

### PR DESCRIPTION
**Pull request by Riey**
###### Motivation for this change

Resolve #116315

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/rtl88x2bu/default.nix [hunk 1]
```diff
@@ -2,23 +2,15 @@
...
     repo = "rtl88x2BU";
-    rev = "a1c53f43fb9995fbe3ad26567079d6384626d350";
-    sha256 = "1cby66jg511zxs1i535mflafhryla9764mnrzacxppimxpancv3s";
+    rev = "48e7c19c92a77554403e1347447f8e2cfd780228";
+    sha256 = "0nw2kgblpq6qlr43gbfxqvq0c83664f4czfwzsyfjr47rj00iyq7";
   };
 
-  patches = [
-    # https://github.com/cilynx/rtl88x2bu/pull/58
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/rtl88x2bu/default.nix:16]: Please remove fetchpatch from inputs.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118792` at e8323955 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.rtl88x2bu</li>
    <li>linuxPackages_hardkernel_4_14.rtl88x2bu</li>
    <li>linuxPackages_hardkernel_latest.rtl88x2bu</li>
    <li>linuxPackages_latest-libre.rtl88x2bu</li>
    <li>linuxPackages_testing_bcachefs.rtl88x2bu</li>
    <li>linuxPackages_xanmod.rtl88x2bu</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/1ebd92bed1fcf299b3abf2b7e806a90d">linuxPackages_4_4.rtl88x2bu</a></li>
    <li><a href="https://gist.github.com/b76e4dc3c4ab3de0a9dad675b31878a4">linuxPackages_4_9.rtl88x2bu</a></li>
  </ul>
</details>
<details>
  <summary>9 packages built successfully:</summary>
  <ul>
    <li>linuxPackages.rtl88x2bu (linuxPackages_5_10.rtl88x2bu ,linuxPackages_xen_dom0.rtl88x2bu)</li>
    <li>linuxPackages_4_14.rtl88x2bu</li>
    <li>linuxPackages_4_19.rtl88x2bu</li>
    <li>linuxPackages_5_11.rtl88x2bu (linuxPackages_latest.rtl88x2bu ,linuxPackages_latest_xen_dom0.rtl88x2bu)</li>
    <li>linuxPackages_5_4.rtl88x2bu</li>
    <li>linuxPackages_hardened.rtl88x2bu (linuxPackages_xen_dom0_hardened.rtl88x2bu)</li>
    <li>linuxPackages_latest_hardened.rtl88x2bu (linuxPackages_latest_xen_dom0_hardened.rtl88x2bu)</li>
    <li>linuxPackages_lqx.rtl88x2bu</li>
    <li>linuxPackages_zen.rtl88x2bu</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/rtl88x2bu/default.nix:34:5:
```
   |
34 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by Riey** [comment]
I think 4.4 and 4.9 failure is related #64916

**Comment by Mindavi** [comment]
FYI: license is gpl2Only.

**Pull request merged**